src/test/java/seedu/clinic/logic/commands/UpdateCommandTest.java

```diff
@@ -0,0 +1,225 @@
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public boolean hasSupplier(Supplier supplier) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void deleteSupplier(Supplier target) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void addSupplier(Supplier supplier) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setSupplier(Supplier target, Supplier editedSupplier) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ObservableList<Supplier> getFilteredSupplierList() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void updateFilteredSupplierList(Predicate<Supplier> predicate) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public boolean hasWarehouse(Warehouse warehouse) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void deleteWarehouse(Warehouse target) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void addWarehouse(Warehouse warehouse) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void setWarehouse(Warehouse target, Warehouse editedWarehouse) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public ObservableList<Warehouse> getFilteredWarehouseList() {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
+        public void updateFilteredWarehouseList(Predicate<Warehouse> predicate) {
+            throw new AssertionError("This method should not be called.");
+        }
+    }
+
+    private class ModelStubWithWarehouse extends ModelStub {
+        private Warehouse warehouse;
+        private Clinic clinic = new Clinic();
+
+        ModelStubWithWarehouse(Warehouse warehouse) {
+            requireNonNull(warehouse);
+            this.warehouse = warehouse;
+            clinic.setWarehouses(List.of(warehouse));
+        }
+
+        @Override public ReadOnlyClinic getClinic() {
+            return clinic;
+        }
+
+        @Override
+        public void setWarehouse(Warehouse target, Warehouse editedWarehouse) {
+            this.warehouse = editedWarehouse;
+            clinic.setWarehouses(List.of(editedWarehouse));
+        }
+
+        @Override
+        public void updateFilteredWarehouseList(Predicate<Warehouse> predicate) {
+
+        }
```

Comment:
Just wish to clarify, is the method empty because this method is allowed to be called but there is no specific function this method can do?

Reply:
Yup, its just a dummy function for the stub

Reply:
Alright, noted